sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -946,6 +947,38 @@ object functions {
   def cosh(columnName: String): Column = cosh(Column(columnName))
 
   /**
+   * Adds a number of days to startDate, given values for startDate and days
+   *
+   * @group datetime_funcs
+   * @since 1.5.0
+   */
+  def date_add(startDate: Column, days: Column): Column = DateAdd(startDate.expr, days.expr)
+
+  /**
+   * Adds a number of days to startDate, given column names for startDate and days
+   *
+   * @group datetime_funcs
+   * @since 1.5.0
+   */
+  def date_add(startDate: String, days: String): Column = date_add(Column(startDate), Column(days))
```

Comment:
@davies In those, you have to accept any combination, because they are math functions. I may want to take the power of 2 with the values in my DataFrame column. I also should be able to take the power of my Column by 2. If we can have a case where `f(a, b) == f(b, a)`, we wouldn't need all combinations, but those functions (pow, atan2) don't necessarily have `f(a, b) == f(b, a)`